source/speech.py

```diff
@@ -1130,7 +1130,9 @@ def getControlFieldSpeech(attrs,ancestorAttrs,fieldType,formatConfig=None,extraD
 	else:
 		tableID = None
 
-	roleText=getSpeechTextForProperties(reason=reason,role=role)
+	roleText=attrs.get('roleText')

```

Comment:
What if we changed this to this?
roleText=attrs.get('roleText', lambda:getSpeechTextForProperties(reason=reason,role=role))
That will prevent the function from ever needing called in the roletext case, and removes that if.